### PR TITLE
Confirm tests

### PR DESCRIPTION
**Pull request by MorganLove**
**Testing**

- Adding unit tests for the Confirm component

**Progress**

- Should be almost done, looking for a way to avoid using `any`, and I'm not sure if the snapshots are working right.

### CODE REVIEW

File: client/src/components/Atoms/Confirm.test.tsx [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+jest.mock('react', () => ({
+  ...jest.requireActual('react'),
+  useState: jest.fn(),
+}))
+
+afterEach(() => {
+  jest.clearAllMocks()
+})
+
+describe('Confirm component', () => {
```
Comment (jonahkagan) [inline client/src/components/Atoms/Confirm.test.tsx:33]: I think it might be simpler to test by creating a "consumer" component that mimics the way the `Confirm` component and `useConfirm` hooks interact in their real usage. Here's an example of what that could look like:

```ts
describe('Confirm', () => {
  it('shows a confirmation dialog', async () => {
    const onYesClickMock = jest.fn().mockResolvedValue(true)

    const ConfirmConsumer = () => {
      const { confirm, confirmProps } = useConfirm()
      return (
        <>
          <Button
            onClick={() =>
              confirm({
                title: 'Test Title',
                description: 'Test description',
                onYesClick: onYesClickMock,
              })
            }
          >
            Open Confirm
          </Button>
          <Confirm {...confirmProps} />
        </>
      )
    }
    render(<ConfirmConsumer />)

    userEvent.click(screen.getByRole('button', { name: 'Open Confirm' }))
    const dialog = screen
      .getByRole('heading', { name: 'Test Title' })
      .closest('.bp3-dialog') as HTMLElement
    userEvent.click(within(dialog).getByRole('button', { name: 'Ok' }))
    expect(onYesClickMock).toHaveBeenCalled()
    await waitFor(() => expect(dialog).not.toBeInTheDocument())
    // ... more interactions and assertions ...
  })
})
```

File: client/src/components/Atoms/Confirm.tsx [hunk 2]
```diff
@@ -15,7 +15,7 @@ export const useConfirm = () => {
   // On close, we set options to null.
   const [options, setOptions] = useState<IConfirmOptions | null>(null)
 
-  const confirm = (newOptions: IConfirmOptions) => {
+  const confirm = (newOptions: IConfirmOptions | null) => {
```
Comment (jonahkagan) [inline client/src/components/Atoms/Confirm.tsx:18]: why this change?

File: client/src/components/Atoms/Confirm.test.tsx [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+    userEvent.click(within(dialog).getByRole('button', { name: 'Yes' }))
+
+    await waitFor(() => {
+      expect(
+        screen.queryByText(
+          'This action cannot be undone. Do you want to continue?'
+        )
+      ).not.toBeInTheDocument()
+    })
+  })
```
Comment (jonahkagan) [inline client/src/components/Atoms/Confirm.test.tsx:93]: might be good to check that it calls `onYesClickMock` when you click Yes

File: client/src/components/Atoms/Confirm.tsx (hunk 2 above)
Comment (MorganLove) [reply client/src/components/Atoms/Confirm.tsx:18]: Oops, that was left there from some earlier testing.

**Review by jonahkagan — APPROVED** [review]
